### PR TITLE
feat(jiva): add playbook for snap deletion

### PR DESCRIPTION
**Pull request by utkarshmani1997**
Signed-off-by: Utkarsh Mani Tripathi <utkarsh.tripathi@mayadata.io>

**What this PR does / why we need it**:
This commit add test case for validating data integrity of jiva volume after 
snapshot deletion.
**Special notes for your reviewer**:
- This build may fail for 1.0.0 release.

### CODE REVIEW

File: experiments/functional/snapshot_deletion/jiva/test.yml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+         when: app_ns != 'litmus'
+
+       - name: Checking the status  of test specific namespace.
+         shell: kubectl get ns {{ app_ns }} -o jsonpath='{.status.phase}'
+         args:
+           executable: /bin/bash
+         register: npstatus
+         until: "'Active' in npstatus.stdout"
+         delay: 30
+         retries: 10
```
Comment (gprasath) [inline experiments/functional/snapshot_deletion/jiva/test.yml]: can you use the create_ns util

File: experiments/functional/snapshot_deletion/jiva/test.yml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+         include: "{{ check_replica_cnt_yml }}"
+
+       - set_fact:
+           flag: "Pass"
+
+       - name: Include snapshot delete test
+         include: "{{ snap_delete_yml }}"
+
+       - set_fact:
+           flag: "Pass"
```
Comment (gprasath) [inline experiments/functional/snapshot_deletion/jiva/test.yml]: can we set this at the end of play

File: experiments/functional/snapshot_deletion/jiva/test_vars.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+## TEST-SPECIFIC PARAMS
+test_name: fio-data-integrity
```
Comment (gprasath) [inline experiments/functional/snapshot_deletion/jiva/test_vars.yml]: change the test name to fio-di-delete-jiva-snapshots

File: experiments/functional/snapshot_deletion/jiva/test_vars.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+fio_write_yml: fio-write.yml
+fio_read_yml: fio-read.yml
+replica_restart_yml: test_replica_restart.yml
+snap_delete_yml: snapshot_delete_test.yml
+check_replica_cnt_yml: check_replica_count.yml
+app_ns: "{{ lookup('env','FIO_NAMESPACE') }}"
+data_check: "{{ lookup('env','DATA_CHECK') }}"
+pvc_yml: pvc.yml
+operator_ns: "openebs"
```
Comment (gprasath) [inline experiments/functional/snapshot_deletion/jiva/test_vars.yml]: this is not used

File: experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  set_fact:
+    replica_name_three: "{{ replica_pod.stdout_lines[2].split()[0] }}"
+
+- name: Delete replica with latest creation timestamp
+  shell: >
+    kubectl delete pod {{ replica_name_three }} -n {{ app_ns }}
+  args:
+    executable: /bin/bash
+
+- name: Pause for some time for a new replica to come up
```
Comment (shashank855) [inline experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml]: There should be comment explaining the reason for this pause.

File: experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+- name: Get the name of replica pods
+  shell: >
+    kubectl get pods -l openebs.io/replica=jiva-replica,openebs.io/persistent-volume={{ pv_name }} -n {{ app_ns }} --sort-by=.metadata.creationTimestamp -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}}'
+  args:
+    executable: /bin/bash
+  register: replica_pod
+
+- name: Set the replica pod names to variables
```
[the remark this answers is not in the document]
Comment (utkarshmani1997) [reply experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml]: sounds good

File: experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml (hunk 5 above)
Comment (utkarshmani1997) [reply experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml]: sounds good

File: experiments/functional/snapshot_deletion/jiva/run_litmus_test.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            value: "4096m"
+
+          - name: FIO_TESTRUN_PERIOD
+            value: "1000"
+
+          - name: DATA_CHECK #data_integrity for jiva  #data_accesibility for cstor
+            value: "data_integrity"
+
+        command: ["/bin/bash"]
+        args: ["-c", "ansible-playbook ./experiments/functional/snapshot_deletion/jiva/test.yml -i /etc/ansible/hosts -vvv; exit 0"]
```
Comment (shashank855) [inline experiments/functional/snapshot_deletion/jiva/run_litmus_test.yml]: Can you reduce verbosity by 1.

Comment (utkarshmani1997) [reply experiments/functional/snapshot_deletion/jiva/run_litmus_test.yml]: ok

**Pull request merged**